In [2]:
import polars as pl
import pandas as pd
import geopandas as gpd
import datetime

import dash
from dash import dcc, html
from dash.dependencies import Input, Output, State
import dash_bootstrap_components as dbc

import plotly.express as px
import plotly.graph_objects as go
import plotly.figure_factory as ff

from src.procedures import fetch_last_data, generate_density_map
from dotenv import dotenv_values

config = dotenv_values("../.env")

In [24]:
# Retrieves the most recently updated data from the database
value = 7

CONNECTION_URI = config.get("CONNECTION_URI")
query = f"""
    SELECT latitude, longitude, acq_date, acq_time, confidence, frp, brightness, second_adm, first_adm
    FROM processed_viirs 
    WHERE acq_date > CURRENT_DATE - INTERVAL '{value} day'"""

processed_viirs = fetch_last_data(query=query, uri_connection=CONNECTION_URI)
df_viirs = processed_viirs.to_pandas()

df_viirs = df_viirs.rename(columns={"frp":"Fire Radiative Power", "second_adm": "Kabupaten/Kota", "first_adm":"Provinsi",
                                  "acq_date":"Tanggal", "confidence":"Confidence", "brightness":"Brightness"})

df_viirs

latitude   longitude    Tanggal  acq_time Confidence  \
0        1.09626  101.001694 2023-06-19      1913    Nominal   
1       -1.24629  116.813202 2023-06-28      1804    Nominal   
2        0.08398  117.484352 2023-07-01      1707    Nominal   
3        0.84778  101.391228 2023-07-16      1906    Nominal   
4       -6.49370  139.341782 2023-07-17       423    Nominal   
...          ...         ...        ...       ...        ...   
114519   5.05186   96.616478 2023-09-12       640    Nominal   
114520   5.27238   95.432167 2023-09-12       640    Nominal   
114521   5.27246   95.458138 2023-09-12       640    Nominal   
114522   5.27366   95.458389 2023-09-12       640    Nominal   
114523   5.31701   95.551788 2023-09-12       640    Nominal   

        Fire Radiative Power  Brightness Kabupaten/Kota Provinsi  
0                       1.29  304.029999           Siak     Riau  
1                       4.93  328.649994           None     None  
2                       1.86  304.059998           None     None  
3                       0.71  303.269989           Siak     Riau  
4                       6.98  339.970001          Mappi    Papua  
...                      ...         ...            ...      ...  
114519                  7.07  337.929993        Bireuen     Aceh  
114520                  2.72  326.170013     Aceh Besar     Aceh  
114521                  1.35  327.459991     Aceh Besar     Aceh  
114522                  3.79  330.480011     Aceh Besar     Aceh  
114523                  1.49  335.750000     Aceh Besar     Aceh  

[114524 rows x 9 columns]

In [ ]:
# df_viirs["Tanggal"] = df_viirs["Tanggal"].astype(str)

# hover_dict = {"latitude":False, "longitude":False, "Tanggal":True, "acq_time":False, 
#               "Confidence":True,"Fire Radiative Power":True, "Kabupaten/Kota":True, 
#               "Provinsi":False, "Brightness":True}


# map_fig = px.density_mapbox(df_viirs, lat="latitude", lon="longitude", z="Brightness",
#                              radius=1, hover_name="Provinsi",
#                              hover_data=hover_dict,
#                             center=dict(lat=-2.5, lon=118), zoom=3.7, color_continuous_scale="matter_r", 
#                             mapbox_style="open-street-map", template="plotly_white"
#                             )

# map_fig.update_layout(autosize=False,width=1200,height=400)
# map_fig.update_layout(margin={"r":1,"t":1,"l":1,"b":1})
# # map_fig.update_layout({'plot_bgcolor': 'rgba(0, 0, 0, 0)','paper_bgcolor': 'rgba(0, 0, 0, 0)',})
# map_fig.update_coloraxes(showscale=True, colorbar=dict(len=0.75, title="Brightness", thickness=15))

# map_fig


In [22]:
def create_line_chart(data: json):


    line_df = df[df["Provinsi"]==province]

    # Upsample to daily frequency and count the number of fires in each day
    line_df = line_df.resample('D', on='Tanggal')['Fire Radiative Power'].count()

    fig = px.line(line_df, x=line_df.index, y=line_df.values,
            labels={"y":"<b>Jumlah Titik Api</b>", "Tanggal":""},template="plotly_white")

    fig.update_layout(autosize=True,width=700,height=400)
#     fig.update_layout(margin={"r":0,"t":1,"l":0,"b":0})
    fig.update_layout(title="Jumlah Titik Api Harian yang Terdeteksi", title_font_size=16)
    fig.update_traces(line_color='indianred')
    fig.update_layout({'plot_bgcolor': 'rgba(0, 0, 0, 0)','paper_bgcolor': 'rgba(0, 0, 0, 0)',})
    fig.update_yaxes(title_font=dict(size=12))
    fig.update_layout(xaxis_showgrid=False, yaxis_showgrid=False)

    return fig

In [23]:
line_2022 = create_line_chart(df_viirs, "Jawa Timur")
line_2022

c:\Users\PF2L6BL6\Documents\alvin\personal-works\data-engineering-projects\integrated-active-fire-and-polution-dashboard\.venv-pm25\Lib\site-packages\_plotly_utils\basevalidators.py:105: FutureWarning:

The behavior of DatetimeProperties.to_pydatetime is deprecated, in a future version this will return a Series containing python datetime objects instead of an ndarray. To retain the old behavior, call `np.array` on the result



In [ ]:
# WHERE acq_date > CURRENT_DATE

In [7]:
CONNECTION_URI = config.get("CONNECTION_URI")
query = f"""
    SELECT *
    FROM articles
    WHERE published_date = CURRENT_DATE"""

articles = fetch_last_data(query=query, uri_connection=CONNECTION_URI)
articles = articles.to_pandas()

In [8]:
articles

id         keywords                                              title  \
0   833  kebakaran hutan  Alasan Kebakaran Hutan Sulit Untuk Dipadamkan ...   
1   834     polusi udara  Jaga Kesehatan Keluarga, Hindari Bahaya Polusi...   
2   824  kebakaran hutan  Imbas Kebakaran Hutan dan Lahan di Ketapang, P...   
3   825     polusi udara  Memperingati Hari Ozon Internasional: Penyadar...   
4   826     polusi udara  5 Tips untuk Menghadapi Kualitas Udara yang Bu...   
5   835     polusi udara  Kemelut Polusi Udara Jakarta: Inkonsistensi Da...   
6   836  kebakaran hutan  Ekonomi Hijau dan Kebakaran Hutan | BALIPOST.c...   
7   827     polusi udara  Mobilitas Warga Jangan Jadi Kambing Hitam Polu...   
8   828  kebakaran hutan  Foto: Kebakaran Hutan dan Lahan Gunung Bromo B...   
9   829  kebakaran hutan  Proses Pemadaman Kebakaran Hutan dan Lahan di ...   
10  830     polusi udara  Muhadjir Effendy Ajak Warga Kurangi Polusi Uda...   
11  831  kebakaran hutan  Peduli Lingkungan, Babinsa Sosialisasikan Sisk...   
12  832  kebakaran hutan  WO Prewedding Ancam Polisikan Pengelola Bromo ...   
13  837     polusi udara  Menjaga Kesehatan Ditengah Kondisi Buruknya Po...   
14  900  kebakaran hutan  Alasan Kebakaran Hutan Sulit Untuk Dipadamkan ...   
15  901     polusi udara  Jaga Kesehatan Keluarga, Hindari Bahaya Polusi...   
16  902     polusi udara  Kemelut Polusi Udara Jakarta: Inkonsistensi Da...   
17  903  kebakaran hutan   Ekonomi Hijau dan Kebakaran Hutan - BALIPOST.com   
18  891     polusi udara  5 Tips untuk Menghadapi Kualitas Udara yang Bu...   
19  899  kebakaran lahan  Asap Kebakaran Lahan Selimuti Jalan Lintas Sum...   
20  892     polusi udara  Mobilitas Warga Jangan Jadi Kambing Hitam Polu...   
21  893  kebakaran hutan  Proses Pemadaman Kebakaran Hutan dan Lahan di ...   
22  894     polusi udara  Muhadjir Effendy Ajak Warga Kurangi Polusi Uda...   
23  895  kebakaran hutan  Peduli Lingkungan, Babinsa Sosialisasikan Sisk...   
24  896  kebakaran lahan  Jembatan Tol Cisumdawu Nyaris Terbakar, Meremb...   
25  897  kebakaran hutan  WO Prewedding Ancam Polisikan Pengelola Bromo ...   
26  898  kebakaran lahan  WO Prewedding Ancam Polisikan Pengelola Bromo ...   
27  919  kebakaran hutan  Alasan Kebakaran Hutan Sulit Untuk Dipadamkan ...   
28  920     polusi udara  Jaga Kesehatan Keluarga, Hindari Bahaya Polusi...   
29  921  kebakaran hutan  Imbas Kebakaran Hutan dan Lahan di Ketapang, P...   
30  922     polusi udara  Memperingati Hari Ozon Internasional: Penyadar...   
31  923     polusi udara  5 Tips untuk Menghadapi Kualitas Udara yang Bu...   
32  924     polusi udara  Kemelut Polusi Udara Jakarta: Inkonsistensi Da...   
33  925  kebakaran hutan  Ekonomi Hijau dan Kebakaran Hutan | BALIPOST.c...   
34  931     polusi udara  Menjaga Kesehatan Ditengah Kondisi Buruknya Po...   
35  926     polusi udara  Mobilitas Warga Jangan Jadi Kambing Hitam Polu...   
36  927  kebakaran hutan  Proses Pemadaman Kebakaran Hutan dan Lahan di ...   
37  928     polusi udara  Muhadjir Effendy Ajak Warga Kurangi Polusi Uda...   
38  929  kebakaran hutan  Peduli Lingkungan, Babinsa Sosialisasikan Sisk...   
39  930  kebakaran hutan  WO Prewedding Ancam Polisikan Pengelola Bromo ...   

                                         article_text  \
0   Kebakaran hutan sering terjadi saat musim kema...   
1   Krjogja.com - Jakarta - Polusi udara di perkot...   
2   Sabtu, 16 September 2023 | 13:40 WIBKETAPANG, ...   
3   Sama seperti yang diungkapkan oleh Soemarwoto ...   
4   Fimela.com, Jakarta Akhir-akhir ini, polusi ud...   
5   A+ A-Belakangan ini masyarakat Indonesia tenga...   
6   Oleh I Wayan RamanthaEkonomi hijau atau green ...   
7   Liputan6.com, Jakarta Peneliti Institute for D...   
8   Balai Besar Taman Nasional Bromo Tengger Semer...   
9   Sabtu, 16 September 2023 | 11:47 WIBBANJARBARU...   
10  JAKARTA, KOMPAS.com - Menteri Koordinator Pemb...   
11  Leuwisadeng-Dalam upaya mencegah kebakaran hut...   
12  --Wedding Organizer 

In [10]:
image = articles["image"][2]
title = articles["title"][2]
url = articles["url"][2]
publish = articles["published_time"][2]
image

'https://media-origin.kompas.tv/lestari/desktop/img/lestari_biruv2.png'

In [34]:
card = dbc.Card(
    [
        dbc.CardImg(src=image, top=True),
        dbc.CardBody(
            [
                # html.H4(f"{title}", className="card-title"),
                html.P(
                    f"{title}",
                    className="card-text",
                ),
                dbc.Button("Go somewhere", color="primary", href=f"{url}"),
            ]
        ),
    ],
    style={"width": "18rem"},
)

card

Card(children=[CardImg(src='https://akcdn.detik.net.id/community/media/visual/2023/09/05/nikita-willy-1_169.png?w=200&q=90', top=True), CardBody([P(children='Bupati Giri Sebut Kebakaran Hutan di Ponorogo Ada Unsur Kesengajaan - detikJatim', className='card-text'), Button(children='Go somewhere', color='primary')])], style={'width': '18rem'})

In [13]:
card = dbc.Card(
    [
        dbc.Row(
            [
                dbc.Col(
                    dbc.CardImg(
                        src=image,
                        className="img-fluid rounded-start",
                    ),
                    className="col-md-4",
                ),
                dbc.Col(
                    dbc.CardBody(
                        [
                            html.H4(f"{title}", className="card-title"),
                            # html.P(
                            #     "This is a wider card with supporting text "
                            #     "below as a natural lead-in to additional "
                            #     "content. This content is a bit longer.",
                            #     className="card-text",
                            # ),
                            html.Small(
                                f"Published date: {publish}",
                                className="card-text text-muted",
                            ),
                        ]
                    ),
                    className="col-md-8",
                ),
            ],
            className="g-0 d-flex align-items-center",
        )
    ],
    className="mb-3",
    style={"maxWidth": "540px"},
    color="warning",
    outline=True
)

In [12]:
import datetime
import dash
from dash import dcc, html
from dash.dependencies import Input, Output, State
import dash_bootstrap_components as dbc

from src.procedures import generate_density_map, fetch_last_data, generate_line_chart, generate_top_prov, generate_top_kabkot

from dotenv import dotenv_values

config = dotenv_values("../.env")
CONNECTION_URI = config.get("CONNECTION_URI")

In [10]:
CONNECTION_URI = config.get("CONNECTION_URI")
query = """
    WITH RANKED_DATA AS (
    SELECT 
        address, city, province, air_quality_index, category, updated_at, 
        ROW_NUMBER() OVER (PARTITION BY address, city, province ORDER BY updated_at DESC) AS rn
    FROM 
        air_quality_idn
    WHERE 
        DATE(fetched_date) = CURRENT_DATE
    )
    SELECT 
        *
    FROM 
        RANKED_DATA
    WHERE 
        rn = 1
    ORDER BY 
        air_quality_index DESC
"""

air_quality = fetch_last_data(query=query, uri_connection=CONNECTION_URI)
air_quality = air_quality.to_pandas()

air_quality["province"] = air_quality["province"].replace({
    "Dki Jakarta":"DKI Jakarta", "Ntt":"NTT", "Ntb":"NTB", "Nad":"Aceh"
})

air_quality

address               city  \
0                   Simpang Icon City, Kota Palembang          Palembang   
1   PM29+F87, Jl. Boulevard BSD Tim., Lengkong Gud...  Tangerang Selatan   
2                               Kantor Bupati Lama OI          Ogan Ilir   
3   Bantar Gebang [MX2V+2C9, RT.001/RW.005, Ciketi...             Bekasi   
4   4, Jl. Monumen Pancasila Sakti No.1, RT.4/RW.9...            Jakarta   
..                                                ...                ...   
60  Wahno, Abepura, Kota Jayapura, Papua 99225, Ko...           Jayapura   
61  Jl. Jenderal Sudirman No.57, RW.11, Klandasan ...         Balikpapan   
62  Surabaya, Balongsari, Tandes, Kota Surabaya, J...           Surabaya   
63  Ktr. Kecamatan Wanea, Jl. 14 Februari No.72, T...             Manado   
64  Kantor Gubernur NTT, Oebobo, Kota Kupang , Nus...             Kupang   

            province  air_quality_index     category          updated_at  rn  
0   Sumatera Selatan                149  Tidak Sehat 2023-09-17 18:00:00   1  
1             Banten                118  Tidak Sehat 2023-09-17 18:00:00   1  
2   Sumatera Selatan                115  Tidak Sehat 2023-09-17 18:00:00   1  
3         Jawa Barat                113  Tidak Sehat 2023-09-17 18:00:00   1  
4        DKI Jakarta                109  Tidak Sehat 2023-09-17 13:00:00   1  
..               ...                ...          ...                 ...  ..  
60             Papua                 28         Baik 2023-09-14 15:00:00   1  
61  Kalimantan Timur                 27         Baik 2023-09-17 18:00:00   1  
62        Jawa Timur                 26         Baik 2023-09-17 18:00:00   1  
63    Sulawesi Utara                 25         Baik 2023-09-17 18:00:00   1  
64               NTT                 21         Baik 2023-09-17 18:00:00   1  

[65 rows x 7 columns]

In [11]:
air_quality.province.unique()

array(['Sumatera Selatan', 'Banten', 'Jawa Barat', 'DKI Jakarta', 'Jambi',
       'Sulawesi Tengah', 'Lampung', 'Riau', 'Kalimantan Selatan',
       'Maluku Utara', 'Kalimantan Tengah', 'Kalimantan Timur',
       'Kalimantan Barat', 'Jawa Tengah', 'Sumatera Barat', 'Yogyakarta',
       'Bangka Belitung', 'Nusa Tenggara Barat', 'Sulawesi Selatan',
       'Kepulauan Riau', 'Aceh', 'Sulawesi Tenggara', 'Gorontalo',
       'Kalimantan_utara', 'Bali', 'Jawa Timur', 'Sulawesi Barat',
       'Papua', 'Sulawesi Utara', 'NTT'], dtype=object)

In [19]:
# Retrieves the most recently updated data from the database
value = [7, 15, 35]

CONNECTION_URI = config.get("CONNECTION_URI")
query = f"""
    SELECT *
    FROM aggregated_fire_summary 
    WHERE aggregated_date = CURRENT_DATE and day_interval = {value[0]}
    ORDER BY timestampt DESC
    LIMIT 1"""

summary = fetch_last_data(query=query, uri_connection=CONNECTION_URI)
summary2 = summary.to_pandas()

summary2

aggregated_date  day_interval  \
0      2023-09-23             7   

                                         daily_count  \
0  {"columns":["Tanggal","total_fires"],"data":[[...   

                                      province_count summary  \
0  {"columns":["Provinsi","total_fires"],"data":[...    None   

                        timestampt  id  
0 2023-09-23 10:28:24.691368+00:00   1

In [4]:
from langchain.agents import create_pandas_dataframe_agent
from langchain.agents.agent_types import AgentType
from langchain.agents import initialize_agent
from langchain.chat_models import ChatOpenAI

In [22]:
def get_summary_event(data, api_key):
    llm = ChatOpenAI(
        model="gpt-3.5-turbo-0613",
        temperature=0.25,
        openai_api_key=api_key
        )
    
    agent = create_pandas_dataframe_agent(
        llm,
        data,
        verbose=True,
        agent_type=AgentType.OPENAI_FUNCTIONS
        )
    
    
    response = agent(
        """
        As a senior Business Intelligence analyst specializing in environmental concerns, analyze the daily_count and province_count columns from the Indonesia active fire dataframe.
        Provide a comprehensive environmental analysis, considering key trends and potential causes such as Indonesia's dry season (based on the dates), the role of major corporations, or government policies.
        Present your findings in Bahasa Indonesia using a casual, empathetic tone. The analysis should be presented in two medium-length paragraphs, and be sure to critically address possible reasons for these fires.
        example:
        Data terbaru titik api di Indonesia, terdeteksi oleh satelit S-NPP dari 11-17 September 2023, menyoroti tren kebakaran yang memprihatinkan. Pada 15 dan 16 September, tercatat lonjakan signifikan titik api sebanyak 5.275 dan 5.400, mengindikasikan potensi kebakaran hutan dan lahan di beberapa wilayah. Ini bertepatan dengan musim kemarau, meningkatkan potensi risiko kebakaran. Tiga provinsi dengan titik api tertinggi adalah Sumatera Selatan, Nusa Tenggara Timur, dan Papua dengan total 4.507, 4.089, dan 1.748 titik. Kondisi ini menegaskan perlunya intervensi dan optimalisasi pengelolaan kebakaran.
        Beberapa faktor penyebab, antara lain musim kemarau, praktek pembukaan lahan oleh korporasi tanpa tanggung jawab, dan kebijakan pengelolaan hutan dan lahan yang belum optimal dapat mempengaruhi frekuensi kebakaran hutan dan lahan. Untuk menanggulangi hal ini, tindakan pencegahan yang efektif harus diterapkan, termasuk pengawasan ketat aktivitas berisiko, penegakan hukum terhadap pembakaran ilegal, serta edukasi masyarakat dan korporasi tentang pentingnya konservasi. Kebijakan pemerintah dari tingkat daerah ke tingkat pusat yang mendukung pengelolaan hutan berkelanjutan juga esensial demi menjaga kelestarian ekosistem di Indonesia.
        """
        )
    
    data["summary"] = response["output"]

    return data

In [23]:
openai_key = config.get("OPENAI_KEY")
summary2 = get_summary_event(summary2, openai_key)

summary2



> Entering new  chain...
Data kebakaran aktif di Indonesia menunjukkan beberapa temuan penting terkait isu lingkungan. Pertama, terdapat tren peningkatan jumlah kebakaran harian selama periode yang diamati. Data menunjukkan bahwa jumlah kebakaran harian meningkat dari 2.309 pada tanggal 17 September menjadi 6.122 pada tanggal 18 September, dan terus meningkat hingga mencapai 4.622 pada tanggal 22 September. Hal ini mengindikasikan bahwa kebakaran hutan dan lahan menjadi masalah yang semakin serius di Indonesia. Trend ini dapat terkait dengan musim kemarau di Indonesia, yang sering kali meningkatkan risiko kebakaran hutan dan lahan. Selain itu, analisis terhadap data province_count juga mengungkapkan beberapa provinsi yang mengalami jumlah kebakaran yang signifikan. Provinsi dengan jumlah kebakaran tertinggi adalah Nusa Tenggara Timur dengan 4.432 kebakaran, diikuti oleh Kalimantan Tengah dengan 1.678 kebakaran, dan Sumatera Selatan dengan 2.215 kebakaran. Temuan ini menunjukkan bahwa

aggregated_date  day_interval  \
0      2023-09-23             7   

                                         daily_count  \
0  {"columns":["Tanggal","total_fires"],"data":[[...   

                                      province_count  \
0  {"columns":["Provinsi","total_fires"],"data":[...   

                                             summary  \
0  Data kebakaran aktif di Indonesia menunjukkan ...   

                        timestampt  id  
0 2023-09-23 10:28:24.691368+00:00   1

In [26]:
text = summary2["summary"][0]

text

'Data kebakaran aktif di Indonesia menunjukkan beberapa temuan penting terkait isu lingkungan. Pertama, terdapat tren peningkatan jumlah kebakaran harian selama periode yang diamati. Data menunjukkan bahwa jumlah kebakaran harian meningkat dari 2.309 pada tanggal 17 September menjadi 6.122 pada tanggal 18 September, dan terus meningkat hingga mencapai 4.622 pada tanggal 22 September. Hal ini mengindikasikan bahwa kebakaran hutan dan lahan menjadi masalah yang semakin serius di Indonesia. Trend ini dapat terkait dengan musim kemarau di Indonesia, yang sering kali meningkatkan risiko kebakaran hutan dan lahan. Selain itu, analisis terhadap data province_count juga mengungkapkan beberapa provinsi yang mengalami jumlah kebakaran yang signifikan. Provinsi dengan jumlah kebakaran tertinggi adalah Nusa Tenggara Timur dengan 4.432 kebakaran, diikuti oleh Kalimantan Tengah dengan 1.678 kebakaran, dan Sumatera Selatan dengan 2.215 kebakaran. Temuan ini menunjukkan bahwa kebakaran hutan dan lahan

Data kebakaran aktif di Indonesia menunjukkan beberapa temuan penting terkait isu lingkungan. Pertama, terdapat tren peningkatan jumlah kebakaran harian selama periode yang diamati. Data menunjukkan bahwa jumlah kebakaran harian meningkat dari 2.309 pada tanggal 17 September menjadi 6.122 pada tanggal 18 September, dan terus meningkat hingga mencapai 4.622 pada tanggal 22 September. Hal ini mengindikasikan bahwa kebakaran hutan dan lahan menjadi masalah yang semakin serius di Indonesia. Trend ini dapat terkait dengan musim kemarau di Indonesia, yang sering kali meningkatkan risiko kebakaran hutan dan lahan. Selain itu, analisis terhadap data province_count juga mengungkapkan beberapa provinsi yang mengalami jumlah kebakaran yang signifikan. Provinsi dengan jumlah kebakaran tertinggi adalah Nusa Tenggara Timur dengan 4.432 kebakaran, diikuti oleh Kalimantan Tengah dengan 1.678 kebakaran, dan Sumatera Selatan dengan 2.215 kebakaran. Temuan ini menunjukkan bahwa kebakaran hutan dan lahan tidak hanya terjadi di satu wilayah tertentu, tetapi menyebar di berbagai provinsi di Indonesia. Terdapat beberapa faktor yang dapat menjadi penyebab kebakaran hutan dan lahan ini. Salah satunya adalah musim kemarau yang meningkatkan risiko kebakaran. Selama musim kemarau, curah hujan yang rendah dan cuaca yang kering membuat tanah dan vegetasi menjadi sangat mudah terbakar. Selain itu, peran korporasi besar dan kebijakan pemerintah juga dapat mempengaruhi tingkat kebakaran. Misalnya, aktivitas perusahaan pertanian atau perkebunan yang menggunakan metode pembakaran lahan untuk membersihkan lahan dapat meningkatkan risiko kebakaran. Selain itu, kebijakan pemerintah terkait pengelolaan hutan dan lahan juga dapat mempengaruhi tingkat kebakaran. Jika kebijakan tersebut tidak optimal atau tidak diterapkan dengan baik, risiko kebakaran hutan dan lahan dapat meningkat. Dalam rangka mengatasi masalah kebakaran hutan dan lahan ini, penting untuk melakukan tindakan pencegahan yang efektif. Hal ini meliputi peningkatan pengawasan terhadap aktivitas perusahaan yang berpotensi menyebabkan kebakaran, penegakan hukum yang tegas terhadap pelaku pembakaran lahan ilegal, dan peningkatan kesadaran masyarakat akan pentingnya menjaga kelestarian lingkungan. Selain itu, pemerintah juga perlu mengimplementasikan kebijakan yang mendukung pengelolaan hutan dan lahan yang berkelanjutan. Dengan langkah-langkah ini, diharapkan dapat mengurangi risiko kebakaran hutan dan lahan serta menjaga kelestarian lingkungan di Indonesia.'

In [5]:
openai_key = config.get("OPENAI_KEY")

llm = ChatOpenAI(model="gpt-3.5-turbo-0613",
                 temperature=0.25,
                 openai_api_key=openai_key)

agent = create_pandas_dataframe_agent(
    llm,
    summary2,
    verbose=True,
    agent_type=AgentType.OPENAI_FUNCTIONS
)

In [6]:
response = agent("""
                 As a senior Business Intelligence with a focus on environmental issues, given this active fire data in daily_count and province_count column, provide a short, 
                 comprehensive and contextual environmental analysis focused on key trends and areas of concern in Bahasa Indonesia.
                 Example:
                 Data terbaru mengenai kebakaran aktif harian di Indonesia dari tanggal 11 hingga 17 September 2023 menunjukkan beberapa tren dan perhatian lingkungan yang penting. 
                 Paling mencolok, ada lonjakan besar kebakaran aktif pada tanggal 15 dan 16 September, dengan 5.275 dan 5.400 kebakaran. Ini mungkin menandakan meningkatnya kebakaran hutan dan lahan. 
                 Tanggal-tanggal ini sesuai dengan musim kemarau di Indonesia, jadi risiko kebakaran hutan mungkin makin tinggi di masa itu. Sumatra Selatan, Nusa Tenggara Timur, dan Papua punya angka kebakaran yang cukup tinggi, dengan 4.507, 4.089, dan 1.748 kebakaran. 
                 Ini menegaskan pentingnya tindakan cepat untuk mencegah dan mengurangi dampak kebakaran, melindungi ekosistem, dan mengatasi masalah deforestasi yang bisa jadi penyebab kebakaran ini.         
""")



> Entering new  chain...
Data terbaru mengenai kebakaran aktif harian di Indonesia menunjukkan beberapa tren dan perhatian lingkungan yang penting. Pada rentang tanggal 17 hingga 23 September 2023, terdapat lonjakan jumlah kebakaran aktif pada tanggal 18 dan 19 September, dengan masing-masing 6.124 dan 4.675 kebakaran. Hal ini menandakan adanya peningkatan risiko kebakaran hutan dan lahan.

Selain itu, terdapat beberapa provinsi yang memiliki jumlah kebakaran aktif yang cukup tinggi. Provinsi dengan jumlah kebakaran tertinggi adalah Nusa Tenggara Timur dengan 4.432 kebakaran, diikuti oleh Papua dengan 4.622 kebakaran. Provinsi Sumatera Selatan juga memiliki jumlah kebakaran yang signifikan, yaitu sebanyak 2.215 kebakaran.

Tingginya jumlah kebakaran aktif ini menunjukkan adanya masalah serius terkait kebakaran hutan dan lahan di Indonesia. Hal ini dapat menyebabkan kerusakan ekosistem, hilangnya habitat satwa liar, dan polusi udara akibat asap kebakaran. Selain itu, kebakaran hutan d

In [8]:
response["output"]

'Data terbaru mengenai kebakaran aktif harian di Indonesia menunjukkan beberapa tren dan perhatian lingkungan yang penting. Pada rentang tanggal 17 hingga 23 September 2023, terdapat lonjakan jumlah kebakaran aktif pada tanggal 18 dan 19 September, dengan masing-masing 6.124 dan 4.675 kebakaran. Hal ini menandakan adanya peningkatan risiko kebakaran hutan dan lahan.\n\nSelain itu, terdapat beberapa provinsi yang memiliki jumlah kebakaran aktif yang cukup tinggi. Provinsi dengan jumlah kebakaran tertinggi adalah Nusa Tenggara Timur dengan 4.432 kebakaran, diikuti oleh Papua dengan 4.622 kebakaran. Provinsi Sumatera Selatan juga memiliki jumlah kebakaran yang signifikan, yaitu sebanyak 2.215 kebakaran.\n\nTingginya jumlah kebakaran aktif ini menunjukkan adanya masalah serius terkait kebakaran hutan dan lahan di Indonesia. Hal ini dapat menyebabkan kerusakan ekosistem, hilangnya habitat satwa liar, dan polusi udara akibat asap kebakaran. Selain itu, kebakaran hutan dan lahan juga dapat be